baselines/depthfl/README.md

```diff
@@ -0,0 +1,166 @@
+
+
+## About this baseline
+
+****What’s implemented:**** The code in this directory replicates the experiments in DepthFL: Depthwise Federated Learning for Heterogeneous Clients (Kim et al., 2023) for CIFAR100, which proposed the DepthFL algorithm. Concretely, it replicates the results for CIFAR100 dataset in Table 2,3 and 4.
+
+****Datasets:**** CIFAR100 from PyTorch's Torchvision
+
+****Hardware Setup:**** These experiments were run on a server with Nvidia 3090 GPUs. Any machine with 1x 8GB GPU or more would be able to run it in a reasonable amount of time. 
+
+****Contributors:**** Minjae Kim
+
+
+## Experimental Setup
+
+****Task:**** Image Classification
+
+****Model:**** ResNet18
+
+**Dataset:** This baseline only includes the CIFAR100 dataset. By default it will be partitioned into 100 clients following IID distribution. The settings are as follow:
+
+| Dataset | #classes | #partitions | partitioning method |
+| :------ | :---: | :---: | :---: |
+| CIFAR100 | 100 | 100 | IID |
+
+**Training Hyperparameters:**
+The following table shows the main hyperparameters for this baseline with their default value (i.e. the value used if you run `python main.py` directly)
+
+| Description | Default Value |
+| ----------- | ----- |
+| total clients | 100 |
+| local epoch | 5 |
+| batch size | 50 |
+| number of rounds | 1000 |
+| participation ratio | 10% |
+| learning rate | 0.1 |
+| learning rate decay | 0.998 |
+| client resources | {'num_cpus': 1.0, 'num_gpus': 0.5 }|
+| data partition | IID |
+| optimizer | SGD with dynamic regularization |
+| alpha | 0.1 |
+
+
+## Environment Setup
+
+To construct the Python environment follow these steps:
+
+```bash
+# install the base Poetry environment
```

Comment:
I'm thinking it would be better to show the complete process of building the environment. Worth adding something like this:
```
# Set python version
pyenv local 3.10.6
# Tell poetry to use python 3.10
poetry env use 3.10.6
# Install
```